Modules/Sources/JetpackStats/Views/CountryMap/InteractiveMapView.swift

```diff
@@ -0,0 +1,421 @@
+            self.lightStyle = lightStyle
+            self.darkStyle = darkStyle
+        }
+
+        init(tintColor: UIColor) {
+            self.lightStyle = MapStyle(
+                colorAxis: [
+                    tintColor.lightened(by: 0.75),
+                    tintColor
+                ],
+                strokeColor: UIColor(white: 0.8, alpha: 1),
+                fillColor: UIColor(white: 0.94, alpha: 1)
+            )
+            self.darkStyle = MapStyle(
+                colorAxis: [
+                    tintColor.lightened(by: 0.7),
+                    tintColor
+                ],
+                strokeColor: UIColor(white: 0.36, alpha: 1),
+                fillColor: UIColor(white: 0.19, alpha: 1)
+            )
+        }
+    }
+
+    let svgResourceName: String
+    let data: [String: Int]
+    let configuration: Configuration
+    @Binding var selectedCountryCode: String?
+
+    init(
+        svgResourceName: String = "world-map",
+        data: [String: Int],
+        configuration: Configuration,
+        selectedCountryCode: Binding<String?>
+    ) {
+        self.svgResourceName = svgResourceName
+        self.data = data
+        self.configuration = configuration
+        self._selectedCountryCode = selectedCountryCode
+    }
+
+    @State private var processedSVG: String?
+
+    @Environment(\.colorScheme) private var colorScheme
+
+    private struct Parameters: Equatable {
+        let data: [String: Int]
+        let colorScheme: ColorScheme
+    }
+
+    private var parameters: Parameters {
+        Parameters(data: data, colorScheme: colorScheme)
+    }
+
+    var body: some View {
+        ZStack {
+            if let processedSVG {
+                SVGWebView(htmlContent: processedSVG, selectedCountryCode: $selectedCountryCode)
+            }
+        }
+        .task(id: parameters) {
+            await updateMap(parameters: parameters)
+        }
+    }
+
+    @MainActor
+    private func updateMap(parameters: Parameters) async {
+        guard let svgContent = await loadSVG(resourceName: svgResourceName) else {
+            return
+        }
+
+        // Get the style for the current color scheme
+        let baseStyle = parameters.colorScheme == .dark ? configuration.darkStyle : configuration.lightStyle
+
+        // Resolve colors in the current trait collection
+        let traitCollection = UITraitCollection(userInterfaceStyle: parameters.colorScheme == .dark ? .dark : .light)
+        let resolvedStyle = MapStyle(
+            colorAxis: baseStyle.colorAxis.map { $0.resolvedColor(with: traitCollection) },
+            strokeColor: baseStyle.strokeColor.resolvedColor(with: traitCollection),
+            fillColor: baseStyle.fillColor.resolvedColor(with: traitCollection)
+        )
+        let processedSVGContent = await processSVG(
+            svgContent: svgContent,
+            data: parameters.data,
+            style: resolvedStyle
+        )
```

Comment:
You should be able to manipulate the SVG styles using JavaScript in the web view? Considering the SVG is already parsed as DOM, it might be more straightforward to do it in JavaScript, instead of searching and replacing strings in Swift.